lisp/ess.el

```diff
@@ -333,41 +318,17 @@ Variables controlling indentation style:
     Non-nil means distinguish between #, ##, and ### for indentation.
 
 Furthermore, \\[ess-set-style] command enables you to set up predefined ess-mode
-indentation style. At present, predefined style are `BSD', `GNU', `K&R', `C++',
-`CLB' (quoted from C language style)."
-  (setq alist (or alist
-                  (buffer-local-value 'ess-local-customize-alist (current-buffer))
-                  (error "Customise alist is not specified, nor  ess-local-customize-alist is set")))
-  (unless is-derived
-    (kill-all-local-variables)) ;; NOTICE THIS! *** NOTICE THIS! *** NOTICE THIS! ***
-  (ess-setq-vars-local alist)
-  (ess-write-to-dribble-buffer
-   (format "(ess-mode-1): ess-language=%s, ess-dialect=%s buf=%s \n"
-           ess-language
-           ess-dialect
-           (current-buffer)))
-  ;; (ess-write-to-dribble-buffer
-  ;;  (format "(ess-mode-1.2): ess-process=%s \n"
-  ;;   (ess-local-process-name ess-local-process-name "none")))
-  (ess-write-to-dribble-buffer
-   (format "(ess-mode-1.5): alist=%s \n" alist))
-  (unless is-derived
-    (setq major-mode 'ess-mode)
-    (setq mode-name (concat "ESS[" (or ess-dialect ess-language) "]")))
-  ;; The following line does the next 20 or so :-).
-  (ess-write-to-dribble-buffer
-   (format "(ess-mode-1.6): editing-alist=%s \n"
-           ess-mode-editing-alist))
-  (ess-setq-vars-local ess-mode-editing-alist)
+indentation style. See `ess-style-alist' for predefined styles."
+  ;; TODO: get rid of these and rely on modes to set variables properly
```

Comment:
Yes! I'm not sure what's the best way to do this. Do we need mode-specific variables? Do we need inheritance across the mode hierarchy? Do we need a different approach for user-facing variables and private variables?

Reply:
We should be setting variables in the "highest" mode and using inheritance for most of this.

Reply:
Have you looked into CEDET by the way? They implement mode-local functions and variables. I think this was one of the inspiration for the mode generics that @vspinu implemented.